src/edu/washington/escience/myria/daemon/MyriaDriverLauncher.java

```diff
@@ -212,9 +214,60 @@ private static int getDriverMemory(final Configuration conf) throws InjectionExc
   }
 
   /**
+   * Prints the expected configuration, compared to the actual configuration given.
+   * Used for logging command line arguments.
+   */
+  private static void logStartupMessage(Class<? extends Name<?>>[] classes, Injector inj) {
+    final StringBuilder sb = new StringBuilder("MyriaDriverLauncher configuration:\n");
+    for (int i = 0; i < classes.length; i++) {
+      final Class<? extends Name<?>> c = classes[i];
+      final NamedParameter annotation = c.getAnnotation(NamedParameter.class);
+
+      final String fullName = c.getName();
+      final String simpleName = c.getSimpleName();
+      final String shortName = annotation.short_name();
+      final String doc = annotation.doc();
+      final String defaultVal;
+      {
+        if (!annotation.default_value().equals(NamedParameter.REEF_UNINITIALIZED_VALUE))
+          defaultVal = annotation.default_value();
+        else if (!annotation.default_class().equals(Void.class))
+          defaultVal = annotation.default_class().getSimpleName();
+        else if (annotation.default_values().length > 0)
+          defaultVal = Arrays.toString(annotation.default_values());
+        else if (annotation.default_classes().length > 0) {
+          Class<?>[] default_classes = annotation.default_classes();
```

Comment:
This is a pretty clunky, C-style way to create an array of default values. Couldn't you just iterate with a foreach loop over `annotation.default_classes()` and append the results of `getSimpleName()` one-at-a-time to a `StringBuilder`?